src/ostree/ot-builtin-commit.c

```diff
@@ -817,12 +828,64 @@ ostree_builtin_commit (int argc, char **argv, OstreeCommandInvocation *invocatio
             goto out;
         }
 
-#ifndef OSTREE_DISABLE_GPGME
       if (opt_key_ids)
         {
+          /* Initialize crypto system */
+          if (!opt_sign_name)
+            opt_sign_name = "ed25519";
```

Comment:
This is not valid if !HAVE_LIBSODIUM

Reply:
mmm... `ostree_sign_get_by_name()` will not find it and return NULL, so exit from this part but yes, probably default should be "dummy" in that case, but not sure about the defaults.